### DISCUSSION TITLE
Lot of Bugs in .Net Maui tools... In Release mode.

### DISCUSSION THREAD

**Discussion by nextcodelab** [dotnet/maui#10233] [General]
1. On build on release mode the .net standard library cannot find the NetStandLibName/Bin/.NetStandard.2.0/android-arm/NetStandLibName.dll
Why need this lib dll no point another bin location in release mode and not auto generated? I need to copy each build to that android net standard folder....

2. AOT enabled crashes the app I test it on android side... disabling the AOT will not crash..

**Comment by yvdh** [comment]
I second that. I have run into so many problems with MAUI that I have just given up, this is NOT ready for production by a long stretch of the imagination
- issues with resources 
- issues with android devices (real and virtual)
- issues with compilations suddenly failing for no apparent reason with cryptic messages
- configuration is way too complex and illogical
- refactoring just a namespace leads a endless hunt of settings in all sort of files to get it right
- documentation and samples are lacking
- missing visual designer for UI
- clunky debugging
- ...

I will have a second look at it in a year or so ...

**Reply by DuckScapePhilip** [reply]
I agree, I think it works great when you are just running in debug mode on your local phone, but running anything in release is causing some serious issues for me.

**Comment by gsr91** [comment]
.Net MAUI is far from production ready. These days I have been very disappointed and evaluating starting again with xamarin forms since it has many critical bugs.

**Reply by Sebosek** [reply]
Personally, I did the same. Started with MAUI, ending with Xamarin Forms.

**Reply by changguangyu** [reply]
I decided to give up .net totally. MAUI is far from production ready but xamarin will be discontinued soon. (iOS 16 and Android 13 is it's last supported version)
Very disappointed with Microsoft. They droped a stable platform before the successor of it is ready, which put all .Net mobile developers into an awkward situation. If I choose Xamarin, it means that by the 2023.9, I cannot support iOS 17. But I doubt MAUI can be used for production at that time.

**Reply by Sebosek** [reply]
I understand the amount of indignation, however, I still believe that Microsoft will handle the situation. They either extend the Xamarine lifetime and add support for the latest OS versions, even without any new API, or they can see light at the end of the tunnel and believe that they will be able to fix all the reported bugs. However, the question is if they will wait for the .NET 8 release window.
Unfortunately, Microsoft, from my point of view, did a mistake to promote the MAUI, at the beginning as a successor of Xamarin, but that isn't true. MAUI is more and less than that at the same time.
It's not only the mobile device framework, it's more like a hybrid application platform. For some reason, they decided to move mobile development to the second railway and focus on and promote the razor components for mobile development.
Bold idea, taking into account the history of such approaches (Cordova, ionic, ...). However, I still think that the reason is simply to promote razor components and try to get some market slice from SPA framework competitors (React, Angular, Vue, Svelte, ...). However, the current demand is somewhere else, currently, we, in SPA, want to have a rendered component on the server and hydrate the app on the client, or even better make the app "state-able on demand" as Quik does.
But getting back to the topic, for me, it's quite simple to make an app in Xamarin and later on migrate to MAUI, even though it may/will require making changes in build pipelines.

**Reply by Xyncgas** [reply]
> I decided to give up .net totally. 

Don't give up .NET there's F# with fable, there's blazor

**Comment by samirgcofficial** [comment]
Try with iOS release you will regret :) even more when sending to test flight with .net 7.

**Reply by AndreasReitberger** [reply]
I actually have no issues with net7 and releases, at least no criticals. I have my app running in testflight and google play (internal tests). Besides some issues with the keyboard on iOS, the app itself is running. No crashes and issues compared to the debug version running on my local device.

**Reply by Xyncgas** [reply]
> Besides some issues with the keyboard on iOS

Microsoft's framework shouldn't have issues

**Comment by Vikas8881** [comment]
I Agree with @yvdh today on 10-03-2023 I installed all the latest versions dotnet 7 visual studio etc.
I start a new fresh App with MAUI XAML. 
below is my experience  
1. 1st of all it's very hard for me to view the Design of the App. every time the build deploys then check. this process consumes much time.
2. Sometimes deployment takes much time more than 30 minutes. I already submit the issue with evidence (a video). [#13750](https://github.com/dotnet/maui/issues/13750)
3. I try the app in a virtual emulator and in a real android Device. there lot of issues I face just in a very simple App.

Now I am confused to stay the MAUI or not I don't know what they try to make a better tool and SPA for Developers Friendly. Because according to my experience personally, it's not a friendly step to remove the designer preview option.
**I don't think Dotnet Maui is Ready for Production yet.**

**Comment by ihsansfd** [comment]
Microsoft, too many innovations, lack of maintanance and quality :')
